js/src/common/components/ModalManager.tsx

```diff
@@ -13,7 +15,14 @@ interface IModalManagerAttrs {
  * overwrite the previous one.
  */
 export default class ModalManager extends Component<IModalManagerAttrs> {
-  view() {
+  protected focusTrap: FocusTrap | undefined;
+
+  /**
+   * Whether a modal is currently shown by this modal manager.
+   */
+  protected modalShown: boolean = false;
```

Comment:
/cc reviewers @askvortsov1 @SychO9 

What are your opinions on the accessibility modifiers here? Should these be public, private or protected? I went with protected for now, as that gives the best of both worlds, but I do wonder if it should be public instead.

Reply:
I prefer it protected, I don't think we'd be breaking any extensions either.

Reply:
The only thing I thought was if an extension closes a modal without using `animateHide()`, then `modalShown` won't be updated, and the focus trap will still exist and be active.

Reply:
how would that happen ?

Reply:
If they directly called `$('.ModalManager').modal('hide')`, or whatever the right selector is. I can't actually think of any reason why they'd do it, so I'm fine with `protected`.

Reply:
yea, there are so many ways extension devs could do things the wrong way, I don't think we can ever do anything about that except have good docs explaining best practices.

Reply:
Let's keep it as protected. We offer a public API for closing the modal, anything else is explicitly going outside of that public API, and therefore not guarunteed not to break.